storage/remote/client.go

```diff
@@ -198,7 +209,28 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		err = errors.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
 	if httpResp.StatusCode/100 == 5 {
-		return RecoverableError{err}
+		return RecoverableError{err, retryAfter{duration: noRetryAfter}}
+	}
+	if httpResp.StatusCode == 429 {
+		recErr := RecoverableError{err, retryAfter{duration: noRetryAfter}}
+		if after := httpResp.Header.Get("Retry-After"); after != "" {
+			duration, err := time.Parse(time.RFC1123, after)
+			if err == nil {
+				recErr.retryAfter = retryAfter{isDateTime: true, after: duration}
```

Comment:
Rather than the `isDateTime` logic, you could just turn this into a `Duration` using `parsedTime.Sub(time.Now())`. Then logic elsewhere only has to care about handling a `Duration`.